File: server/Models/VaultKeep.cs
namespace keepr.Models;

public class VaultKeep : RepoItem<int>
{
    public int KeepId { get; set; }
    public int VaultId { get; set; }
    public string CreatorId { get; set; }
}

public class VaultKeepProfile : Profile
{
    public int VaultKeepId { get; set; }
    public int VaultId { get; set; }
    public int KeepId { get; set; }
}

File: client/src/models/VaultKeep.js
export class VaultKeep{
    constructor(data){
        this.id = data.id
        this.createdAt = new Date(data.createdAt)
        this.updatedAt = new Date(data.updatedAt)
        this.vaultKeepId = data.vaultKeepId
        this.vaultId = data.vaultId
        this.name = data.name
        this.description = data.description
        this.img = data.img
        this.views = data.views
        this.kept = data.kept
        this.creatorId = data.creatorId
        this.creator = data.creator
    }
}
